visor/init.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2022 Gobalsky Labs Limited
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at https://www.mariadb.com/bsl11.
+//
+// Change Date: 18 months from the later of the date of the first publicly
+// available Distribution of this version of the repository, and 25 June 2022.
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by version 3 or later of the GNU General
+// Public License.
+
+package visor
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"code.vegaprotocol.io/vega/logging"
+	"code.vegaprotocol.io/vega/visor/config"
+	"code.vegaprotocol.io/vega/visor/utils"
+)
+
+func Init(log *logging.Logger, homeFolder string) error {
+	homePath, err := utils.AbsPath(homeFolder)
+	if err != nil {
+		return fmt.Errorf("failed to get absolute path for %q: %w", homeFolder, err)
+	}
+
+	homeExists, err := utils.PathExists(homePath)
+	if err != nil {
+		return fmt.Errorf("failed to check if %q path exists: %w", homePath, err)
+	}
+
+	if homeExists {
+		return fmt.Errorf("home folder %q already exists", homePath)
+	}
+
+	visorConf := config.DefaultVisorConfig(log, homePath)
+
+	log.Info("Initiating genesis folder")
+	if err := initDefaultFolder(visorConf.GenesisFolder(), "genesis"); err != nil {
+		return err
+	}
+
+	log.Info("Initiating upgrade folder")
+	if err := initDefaultFolder(visorConf.UpgradeFolder(), "upgrade"); err != nil {
```

Comment:
is the upgradefolder() used for anything?

Reply:
Yes, that is where the validator can put next binaries and configs for the upgrade.